lib/ejs.js

```diff
@@ -721,75 +712,97 @@ Template.prototype = {
 
     newLineCount = (line.split('\n').length - 1);
 
-    switch (line) {
-    case '<' + d:
-    case '<' + d + '_':
-      this.mode = Template.modes.EVAL;
-      break;
-    case '<' + d + '=':
-      this.mode = Template.modes.ESCAPED;
-      break;
-    case '<' + d + '-':
-      this.mode = Template.modes.RAW;
-      break;
-    case '<' + d + '#':
-      this.mode = Template.modes.COMMENT;
-      break;
-    case '<' + d + d:
-      this.mode = Template.modes.LITERAL;
-      this.source += '    ; __append("' + line.replace('<' + d + d, '<' + d) + '")' + '\n';
-      break;
-    case d + d + '>':
-      this.mode = Template.modes.LITERAL;
-      this.source += '    ; __append("' + line.replace(d + d + '>', d + '>') + '")' + '\n';
-      break;
-    case d + '>':
-    case '-' + d + '>':
-    case '_' + d + '>':
-      if (this.mode == Template.modes.LITERAL) {
-        this._addOutput(line);
+    var isHandled;
```

Comment:
Not a huge deal here, but I generally prefer to hoist all the declarations to the very top, to reflect what the runtime actually does.